### DISCUSSION TITLE
Cannot get zbm.timeout to work

### DISCUSSION THREAD

**Discussion by tomk3003** [zbm-dev/zfsbootmenu#591] [Questions]
I am using the EFI bundle installed as described in 
[https://docs.zfsbootmenu.org/en/v2.3.x/guides/ubuntu/uefi.html#install-zfsbootmenu]()
and have set the command line parameters with zbm-kcl.
zbm.prefer works as expected but the zbm.timeout=4 does not seem to have any effect. It's always 10 seconds.
Here the output of zbm-kcl and efibootmgr:
```
$ efibootmgr -v
BootCurrent: 0003
Timeout: 1 seconds
BootOrder: 0003,0002,0000,0001
Boot0000* ubuntu	HD(1,GPT,3e94ffe9-1be9-44be-91a9-52301a5dccf1,0x800,0x100000)/File(\EFI\UBUNTU\SHIMX64.EFI)
Boot0001* UEFI: Built-in EFI Shell	VenMedia(5023b95c-db26-429b-a648-bd47664c8012)..BO
Boot0002* ZFSBootMenu (Backup)	HD(1,GPT,3e94ffe9-1be9-44be-91a9-52301a5dccf1,0x800,0x100000)/File(\EFI\ZBM\VMLINUZ-BACKUP.EFI)
Boot0003* ZFSBootMenu	HD(1,GPT,3e94ffe9-1be9-44be-91a9-52301a5dccf1,0x800,0x100000)/File(\EFI\ZBM\VMLINUZ.EFI)

$ zbm-kcl /boot/efi/EFI/ZBM/VMLINUZ.EFI
quiet loglevel=0 nomodeset zbm.timeout=4 zbm.prefer=rpool/ROOT/ubuntu_zljq9y

$ zfs get all rpool/ROOT/ubuntu_zljq9y |grep bootfs
rpool/ROOT/ubuntu_zljq9y  com.ubuntu.zsys:bootfs       yes                                                                    local
``` 
What Am I doing wrong?

**Comment by tomk3003** [comment]
Setting the bootfs parameter on rpool again seems to have fixed it. Strange, the output of
`zfs get all rpool/ROOT/ubuntu_zljq9y |grep bootfs`
is exactly as it was before.

**Reply by zdykstra** [reply]
bootfs is a pool property, so it needs to be retrieved with zpool. zbm.prefer expects a pool argument, not a dataset argument.

**Reply by tomk3003** [reply]
Thanks, I corrected the zbm.prefer setting. `zpool get bootfs` arlready showed the correct value.
